doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownParser.java

```diff
@@ -74,48 +80,103 @@
 
     /**
      * Regex that identifies a multimarkdown-style metadata section at the start of the document
+     *
+     * In order to ensure that we have minimal risk of false positives when slurping metadata sections, the
+     * first key in the metadata section must be one of these standard keys or else the entire metadata section is
+     * ignored.
      */
-    private static final String MULTI_MARKDOWN_METADATA_SECTION =
-        "^(((?:[^\\s:][^:]*):(?:.*(?:\r?\n\\p{Blank}+[^\\s].*)*\r?\n))+)(?:\\s*\r?\n)";
+    private static final Pattern METADATA_SECTION_PATTERN = Pattern.compile(
+            "\\A^\\s*"
+            + "(?:title|author|date|address|affiliation|copyright|email|keywords|language|phone|subtitle)"
+            + "\\h*:\\h*\\V*\\h*$\\v+"
+            + "(?:^\\h*[^:\\v]+\\h*:\\h*\\V*\\h*$\\v+)*",
+            Pattern.MULTILINE | Pattern.CASE_INSENSITIVE );
 
     /**
      * Regex that captures the key and value of a multimarkdown-style metadata entry.
      */
-    private static final String MULTI_MARKDOWN_METADATA_ENTRY =
-        "([^\\s:][^:]*):(.*(?:\r?\n\\p{Blank}+[^\\s].*)*)\r?\n";
-
-    /**
-     * In order to ensure that we have minimal risk of false positives when slurping metadata sections, the
-     * first key in the metadata section must be one of these standard keys or else the entire metadata section is
-     * ignored.
-     */
-    private static final String[] STANDARD_METADATA_KEYS =
-        { "title", "author", "date", "address", "affiliation", "copyright", "email", "keywords", "language", "phone",
-            "subtitle" };
+    private static final Pattern METADATA_ENTRY_PATTERN = Pattern.compile(
+            "^\\h*([^:\\v]+?)\\h*:\\h*(\\V*)\\h*$",
+            Pattern.MULTILINE );
 
     /**
      * <p>getType.</p>
      *
      * @return a int.
      */
+    @Override
     public int getType()
     {
         return TXT_TYPE;
     }
 
+    /**
+     * The parser of the HTML produced by Flexmark, that we will
+     * use to convert this HTML to Sink events
+     */
     @Requirement
     private MarkdownHtmlParser parser;
 
+    /**
+     * Flexmark's Markdown parser (one static instance fits all)
+     */
+    private static final com.vladsch.flexmark.parser.Parser FLEXMARK_PARSER;
+
+    /**
+     * Flexmark's HTML renderer (its output will be re-parsed and converted to Sink events)
+     */
+    private static final HtmlRenderer FLEXMARK_HTML_RENDERER;
+
+    // Initialize the Flexmark parser and renderer, once and for all
+    static
+    {
+        MutableDataSet flexmarkOptions = new MutableDataSet();
+
+        // Emulate Pegdown's behavior
```

Comment:
What is yours opinion about using this emulation? Should we stay with this or switch to some different profile?

Reply:
Good question.

Flexmark's default behavior is [CommonMark 0.28](https://spec.commonmark.org/0.28/).

Unit tests weren't covering edge cases like Pegdown vs CommonMark, so I was concerned to break people's documentation and I kept the Pegdown profile. It's really a question of balance between risk of breaking existing stuff, and moving forward with something cleaner and properly specified.

I'm personally in favor of moving forward with CommonMark (just like Web developers are happy that we finally got rid of Microsoft Internet Explorer... :-D)

Reply:
Let's try after this PR merge.

Reply:
i would favor move to CommonMark and move master to 1.10.

Reply:
Can we move to CommonMark instead of sticking to ages?

Reply:
Okay, you're right! Guys, fasten your seatbelts, let's do this! :-)

Reply:
I'm removing the Pegdown emulation profile option, but I will keep all of the extensions that used to be enabled. We don't want to lose features.

Also, I'm thinking of enabling the [YAML Extension](https://github.com/vsch/flexmark-java/wiki/Extensions#yaml-front-matter), which is very similar to what we're doing to process metadata, but that will be for another issue.